evalml/tests/component_tests/test_imputer.py

```diff
@@ -521,53 +521,87 @@ def test_imputer_does_not_erase_ww_info():
     mock_transform.call_args[0][0].ww.schema == df_holdout.ww[["b"]].ww.schema
 
 
+def get_int_df():
+    return pd.DataFrame(pd.Series([1, 2, 3], dtype="int64"))
+
+
+def get_float_df():
+    return pd.DataFrame(pd.Series([1.0, 2.0, 4.0], dtype="float"))
+
+
+def get_category_df():
+    return pd.DataFrame(pd.Series(["a", "b", "a"], dtype="category"))
+
+
+def get_bool_df():
+    return pd.DataFrame(pd.Series([True, False, True], dtype=bool))
+
+
+def get_string_df():
+    return pd.DataFrame(
+        pd.Series(
+            ["this will be a natural language column because length", "yay", "hay"],
+            dtype="string",
+        )
+    )
+
+
 @pytest.mark.parametrize(
-    "X_df",
+    "data",
     [
-        pd.DataFrame(pd.Series([1, 2, 3], dtype="Int64")),
-        pd.DataFrame(pd.Series([1.0, 2.0, 4.0], dtype="float")),
-        pd.DataFrame(pd.Series(["a", "b", "a"], dtype="category")),
-        pd.DataFrame(pd.Series([True, False, True], dtype=bool)),
-        pd.DataFrame(
-            pd.Series(
-                ["this will be a natural language column because length", "yay", "hay"],
-                dtype="string",
-            )
-        ),
+        "integer_data",
+        "float_data",
+        "categorical_data",
+        "boolean_data",
     ],
 )
-@pytest.mark.parametrize("has_nan", [True, False])
+@pytest.mark.parametrize(
+    "logical_type", ["Integer", "Double", "Categorical", "NaturalLanguage", "Boolean"]
+)
+@pytest.mark.parametrize("has_nan", ["has_nan", "no_nans"])
```

Comment:
I did it so that the output in either pytest terminal or PyCharm looks like this:

![image](https://user-images.githubusercontent.com/64713315/154332058-ccfc7d9e-8373-4f07-8286-df0b365dea76.png)

The parameters used to just have like "-True-" or "-False-" which made the debugger wonder: "what parameter is failing?"  I refactored them this way so that you could more easily see that data that "has_nan" is buggy rather than seeing that data with "True" was buggy then having to constantly remember that the parameter in that position is "has_nan".